enhancements/windows-containers/windows-node-egress-proxy.md

```diff
@@ -0,0 +1,327 @@
+### Test Plan & Infrastructure Needed
+
+In addition to unit testing individual WMCO packages and controllers, an e2e will be added to the WMCO repo for the
+master/release-4.14 branches. We will leverage the [existing proxy workflow](https://steps.ci.openshift.org/workflow/openshift-e2e-aws-proxy)
+from the release repo and add a job to the WMCO repo's CI config to test with. We will run regression testing to make
+sure the addition of the egress proxy feature does not break existing functionality. If we release a community offering
+with this feature, we will need to add a CI job using a cluster-wide proxy on OKD.
+
+### Open Questions
+
+> 1. Is testing cluster-wide proxy in CI on __only AWS__ okay? This is consistent with what the Linux side does.
+     QE may want to cover all platforms though.
+> 2. Can we use the existing [CI proxy job](https://github.com/openshift/release/tree/master/ci-operator/step-registry/ipi/conf/aws/proxy)
+     to test an HTTPS proxy that requires custom certs to access, or can we only test HTTP and no-proxy using it? If the
+     latter, we'll need to explore how to do this in EC2 and add improvements to the linked workflow.
+> 3. Do user-created Windows workloads need proxy config set in their spec? Does this already happen for Linux
+     counterparts? e.g. setting proxy env vars and mounting the trustedCA ConfigMap as a volume to the deployment spec
+     Regardless, I don't think this should be WMCO's responsibility, so is this on another team or upon the user?
+  <!-- TODO: update non-goals based on the answer to number 3  -->
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+A community version of WMCO 8 or 9 can be released with incremental additions to Windows proxy support
+functionality, giving users an opportunity to get an early preview of the feature using OKD/OCP 4.13 or 4.14.
+It will also allow us to collect feedback to troubleshoot common pain points and learn if there are any shortcomings.
+
+#### Tech Preview -> GA
+
+The feature associated with this enhacement is targeted to land in the offical Red Hat operator version of WMCO 9.0.0
+within OpenShift 4.14 timeframe. The normal WMCO release process will be followed as the functionality described in this
+enhancement is integrated into the product.
+
+An Openshift docs update announcing Windows cluster-wide proxy support will be required as part of GA. The new docs
+should list any Windows-specific info, but linking to existing docs should be enough for overarching proxy/PKI details.
+
+#### Removing a deprecated feature
+
+N/A, as this is a new feature that does not supercede an existing one
+
+### Upgrade / Downgrade Strategy
+
+The relevant upgrade path is from WMCO 8.y.z in OCP 4.13 to WMCO 9.y.z in OCP 4.14. There will be no changes to the
+current WMCO upgrade strategy. Once customers are on WMCO 9.0.0:
+* if they have an existing cluster-wide proxy configured, their Windows nodes will begin utilizing it for egress traffic
```

Comment:
This implies we are supporting this upgrade path i.e. a cluster with proxy enabled running WCMO in an unsupported configuration being upgraded. Do we want to do that?